conjur/cli.py

```diff
@@ -231,6 +292,34 @@ def run_client_action(resource, args):
             # A successful exit is required to prevent the initialization of
             # the Client because the init command does not require the Client
             return
+        elif resource == 'login':
+            credential_data = CredentialData(None, args.name, None)
+            credentials = CredentialsFromFile()
+            login_logic = LoginLogic(credentials)
+            login_controller = LoginController(ssl_verify=args.ssl_verify,
+                                               password=args.password,
+                                               credential_data=credential_data,
+                                               login_logic=login_logic)
+            login_controller.load()
+            return
+        elif resource == 'logout':
+            credentials = CredentialsFromFile()
+            logout_logic = LogoutLogic(credentials)
+
+            logout_controller = LogoutController(ssl_verify=None,
+                                                 logout_logic=logout_logic)
+            logout_controller.remove_credentials()
+            return
+        # pylint: disable=line-too-long
+        if not os.path.exists(DEFAULT_NETRC_FILE) or (os.path.exists(DEFAULT_NETRC_FILE) and os.path.getsize(DEFAULT_NETRC_FILE) == 0):
```

Comment:
I don't understand.
You do login logic for every command?

Reply:
Glad you asked this question because it means I need to add a comment here. 
In the Ruby CLI, when a user executes a command and they are not logged in, we will prompt the user to log in before we execute the command. This is just good UX because we aren't failing, rather telling them there is a prereq
This is what I am doing here. I am check if the either 1. the netrc file exists or if it exists and is blank. If so, this means the user hasn't logged in and I prompt them to do so.

Reply:
I see.
I have two things to say about this:
1. This `if` must go into the `login` logic. It shouldn't rely here, where it doesn't belong. Add your comment above it to explain why you call `login` every time.
2. IMO, this is not necessarily a better UX. other tools tell you you should log in, and fail, instead of logging you in for each command.
But, if this is the UX guideline that you strive to, why not doing the same for `init`?

Reply:
The logic exists for init as well

FYI @InbalZilberman what do you think about the UX